shell/run_shell_cmd_test.go

```diff
@@ -53,3 +53,36 @@ func TestRunShellOutputToStderrAndStdout(t *testing.T) {
 	assert.True(t, strings.Contains(stderr.String(), "Terraform"), "Output directed to stderr")
 	assert.True(t, len(stdout.String()) == 0, "No output to stdout")
 }
+
+func TestGitLevelTopDirCaching(t *testing.T) {
+	t.Parallel()
+
+	terragruntOptions, err := options.NewTerragruntOptionsForTest("")
+	assert.Nil(t, err, "Unexpected error creating NewTerragruntOptionsForTest: %v", err)
+
+	path := "./"
+
+	cmd, err := RunShellCommandWithOutput(terragruntOptions, path, true, false, "git", "rev-parse", "--show-toplevel")
+	assert.NoError(t, err)
+	expectedResult := strings.TrimSpace(cmd.Stdout)
+
+	actualResult, err := GitTopLevelDir(terragruntOptions, path)
+	assert.Nil(t, err, "Unexpected error executing GitTopLevelDir: %v", err)
```

Comment:
I was wondering why can't be used `assert.NoError()`

Reply:
Exchanged